src/server/getters/get-markets-awaiting-designated-reporting.ts

```diff
@@ -1,14 +1,14 @@
 import { each } from "async";
 import * as Knex from "knex";
 import { Address, MarketsRow, UIMarketInfo, UIMarketsInfo, ErrorCallback } from "../../types";
-import { reshapeMarketsRowToUIMarketInfo } from "./get-market-info";
+import { reshapeMarketsRowToUIMarketInfo, getMarketsWithReportingState } from "./get-market-info";
 import { sortDirection } from "../../utils/sort-direction";
 
 // Look up all markets that are currently awaiting designated (automated) reporting.
 // Should accept a designatedReporterAddress parameter that filters by designated reporter address.
 export function getMarketsAwaitingDesignatedReporting(db: Knex, designatedReporter: Address|null, sortBy: string|null|undefined, isSortDescending: boolean|null|undefined, limit: number|null|undefined, offset: number|null|undefined, callback: (err?: Error|null, result?: any) => void): void {
   const queryData: {} = (designatedReporter != null) ? { designatedReporter } : {};
-  let query = db("markets").where(queryData).whereNull("marketStateID");
+  let query = getMarketsWithReportingState(db).where(queryData).whereNull("markets.marketStateID");
```

Comment:
Hm, possibly would be clearer to write this as --
```
let query = getMarketsWithReportingState(db).whereNull("markets.marketStateID");
if (designatedReporter != null) query = query.where({ designatedReporter });
```
What do you think?

Reply:
That's much better. I think you could argue the queryData object works if you can do the build in a single line, but if you're making it a variable anyway, this is significantly better